File: src/js/q11.js
const cache = [];

function fib(n) {
    if (cache[n] == null) {
        if ((n === 0) || n === 1) {
            cache[n] = 1;
        } else {
            cache[n] = fib(n - 1) + fib(n - 2);
        }
    }
    return cache[n];
}

let count = 0;
let n = 1;
while (count < 10) {
    let result = fib(n);
    let sum = 0;
    for (let item of result.toString().split("")) {
        sum += parseInt(item);
    }
    if (result % sum === 0) {
        console.log(result);
        count++;
    }
    n++;
}

File: src/py/q11.py
if __name__ == '__main__':
    memo = {}


    def fib(n):
        if n in memo:
            return memo[n]
        if n == 0 or n == 1:
            memo[n] = 1
        else:
            memo[n] = fib(n - 1) + fib(n - 2)
        return memo[n]


    a = b = 1
    count = 0
    while count < 11:
        c = a + b
        for e in str(c):
            sum += int(e)
        if c % sum == 0:
            print(c)
            count += 1
        a, b = b, c
